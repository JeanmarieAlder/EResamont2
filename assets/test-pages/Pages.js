/* istanbul ignore file */

export const navigationNoChildPlainText = {
  state: {
    params: {
      id: 103,
      updated: 1511353009453,
      unactivated: false,
      icon: "Information",
      position: 1,
      parentId: 88,
      pages_lang: [
        {
          id: 40,
          language: {
            id: 2,
            language: "Italian"
          },
          title: "Francia",
          text: "",
          plaintext:
            "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> Itailan page</p>"
        },
        {
          id: 39,
          language: {
            id: 1,
            language: "French"
          },
          title: "France",
          text: "",
          plaintext:
            "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> French page</p>"
        },
        {
          id: 98,
          language: {
            id: 3,
            language: "English"
          },
          title: "France",
          text: "",
          plaintext:
            "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> English page</p>"
        }
      ],
      children: [],
      deleted: false
    }
  }
};

export const navigationId95 = {
  state: {
    params: {
      id: 95,
      updated: 1511353009453,
      unactivated: false,
      icon: "Information",
      position: 1,
      parentId: 88,
      pages_lang: [
        {
          id: 40,
          language: {
            id: 2,
            language: "Italian"
          },
          title: "Francia",
          text: "",
          plaintext:
            "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> Itailan page</p>"
        },
        {
          id: 39,
          language: {
            id: 1,
            language: "French"
          },
          title: "France",
          text: "",
          plaintext:
            "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> French page</p>"
        },
        {
          id: 98,
          language: {
            id: 3,
            language: "English"
          },
          title: "France",
          text: "",
          plaintext:
            "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> English page</p>"
        }
      ],
      children: [],
      deleted: false
    }
  }
};
export const navigationId100 = {
  state: {
    params: {
      id: 100,
      updated: 1511353009453,
      unactivated: false,
      icon: "Information",
      position: 1,
      parentId: 88,
      pages_lang: [
        {
          id: 40,
          language: {
            id: 2,
            language: "Italian"
          },
          title: "Francia",
          text: "",
          plaintext:
            "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> Itailan page</p>"
        },
        {
          id: 39,
          language: {
            id: 1,
            language: "French"
          },
          title: "France",
          text: "",
          plaintext:
            "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> French page</p>"
        },
        {
          id: 98,
          language: {
            id: 3,
            language: "English"
          },
          title: "France",
          text: "",
          plaintext:
            "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> English page</p>"
        }
      ],
      children: [],
      deleted: false
    }
  }
};

export const navigationNoChildText = {
  state: {
    params: {
      id: 103,
      updated: 1511353009453,
      unactivated: false,
      icon: "Information",
      position: 1,
      parentId: 88,
      pages_lang: [
        {
          id: 40,
          language: {
            id: 2,
            language: "Italian"
          },
          title: "Francia",
          text:
            "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> Italian page</p>",
          plaintext: ""
        },
        {
          id: 39,
          language: {
            id: 1,
            language: "French"
          },
          title: "France",
          text:
            "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> French page</p>",
          plaintext: ""
        },
        {
          id: 98,
          language: {
            id: 3,
            language: "English"
          },
          title: "France",
          text:
            "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> English page</p>",
          plaintext: ""
        }
      ],
      children: [],
      deleted: false
    }
  }
};

export const navigationWithChildrenNoText = {
  push: () => {},
  state: {
    params: {
      id: 88,
      updated: 1521799301078,
      unactivated: false,
      icon: "fa-phone",
      position: 4,
      parentId: 0,
      pages_lang: [
        {
          id: 9,
          language: { id: 1, language: "French" },
          title: "Consultation de médecine de montagne",
          text: "",
          plaintext: ""
        },
        {
          id: 10,
          language: { id: 2, language: "Italian" },
          title: "Ambulatori di medicina di montagna",
          text: "",
          plaintext: ""
        },
        {
          id: 89,
          language: { id: 3, language: "English" },
          title: "Mountain medicine consultation",
          text: "",
          plaintext: ""
        }
      ],
      children: [
        {
          id: 103,
          updated: 1511353009453,
          unactivated: false,
          icon: "Information",
          position: 1,
          parentId: 88,
          pages_lang: [
            {
              id: 40,
              language: { id: 2, language: "Italian" },
              title: "Francia",
              text: "<p>Page in italian</p>",
              plaintext: ""
            },
            {
              id: 39,
              language: { id: 1, language: "French" },
              title: "France",
              text: "<p>Page in French</p>",
              plaintext: ""
            },
            {
              id: 98,
              language: { id: 3, language: "English" },
              title: "France",
              text: "<p>Page in English</p>",
              plaintext: ""
            }
          ],
          children: [],
          deleted: false
        },
        {
          id: 90,
          updated: 1499077163111,
          unactivated: false,
          icon: "Information",
          position: 2,
          parentId: 88,
          pages_lang: [
            {
              id: 13,
              language: { id: 1, language: "French" },
              title: "Italie",
              text: "<p>Page in French</p>",
              plaintext: ""
            },
            {
              id: 14,
              language: { id: 2, language: "Italian" },
              title: "Italia",
              text: "<p>Page in Italian</p>",
              plaintext: ""
            },
            {
              id: 91,
              language: { id: 3, language: "English" },
              title: "Italy",
              text: "<p>Page in English</p>",
              plaintext: ""
            }
          ],
          children: [],
          deleted: false
        },
        {
          id: 117,
          updated: 1499077261455,
          unactivated: false,
          icon: "Information",
          position: 3,
          parentId: 88,
          pages_lang: [
            {
              id: 67,
              language: { id: 1, language: "French" },
              title: "Suisse",
              text: "<p>Page in French</p>",
              plaintext: ""
            },
            {
              id: 68,
              language: { id: 2, language: "Italian" },
              title: "Svizzera",
              text: "<p>Page in Italian</p>",
              plaintext: ""
            },
            {
              id: 113,
              language: { id: 3, language: "English" },
              title: "Switzerland",
              text: "<p>Page in English</p>",
              plaintext: ""
            }
          ],
          children: [],
          deleted: false
        },
        {
          id: 115,
          updated: 1499240819120,
          unactivated: false,
          icon: "Information",
          position: 4,
          parentId: 88,
          pages_lang: [
            {
              id: 64,
              language: { id: 2, language: "Italian" },
              title: "In Europa",
              text: "<p>Page in Italian</p>",
              plaintext: ""
            },
            {
              id: 63,
              language: { id: 1, language: "French" },
              title: "En Europe",
              text: "<p>Page in French</p>",
              plaintext: ""
            },
            {
              id: 112,
              language: { id: 3, language: "English" },
              title: "In Europe",
              text: "<p>Page in English</p>",
              plaintext: ""
            }
          ],
          children: [],
          deleted: false
        }
      ],
      deleted: false
    }
  }
};

export const navigationWithChildrenAndText = {
  push: () => {},
  state: {
    params: {
      id: 126,
      updated: 1521799301121,
      unactivated: false,
      icon: "fa-comment",
      position: 5,
      parentId: 0,
      pages_lang: [
        {
          id: 129,
          language: {
            id: 2,
            language: "Italian"
          },
          title: "Questionari",
          text: "<p>Text in Italian</p>",
          plaintext: ""
        },
        {
          id: 127,
          language: {
            id: 3,
            language: "English"
          },
          title: "Questionnaires",
          text: "<p>Text in English</p>",
          plaintext: ""
        },
        {
          id: 128,
          language: {
            id: 1,
            language: "French"
          },
          title: "Questionnaires",
          text: "",
          plaintext:
            "<p>Text in French.  Also it's located in plaintext field.</p>"
        }
      ],
      children: [
        {
          id: 128,
          updated: 1507220001025,
          unactivated: false,
          icon: "fa-pencil",
          position: 1,
          parentId: 126,
          pages_lang: [
            {
              id: 134,
              language: {
                id: 1,
                language: "French"
              },
              title: "Questionnaire de satisfaction",
              text: "",
              plaintext: "<p>Page in French.</p>"
            },
            {
              id: 135,
              language: {
                id: 2,
                language: "Italian"
              },
              title: "Sondaggio di soddisfazione",
              text: "<p>Page in Italian</p>",
              plaintext: ""
            },
            {
              id: 133,
              language: {
                id: 3,
                language: "English"
              },
              title: "Questionnaire of satisfaction",
              text: "<p>Page in English</p>",
              plaintext: ""
            }
          ],
          children: [],
          deleted: false
        },
        {
          id: 124,
          updated: 1507220001041,
          unactivated: false,
          icon: "fa-pencil",
          position: 2,
          parentId: 126,
          pages_lang: [
            {
              id: 83,
              language: {
                id: 3,
                language: "English"
              },
              title: "Mountaineer General Health Survey",
              text: "",
              plaintext: "<p>Page in English</p>"
            },
            {
              id: 82,
              language: {
                id: 2,
                language: "Italian"
              },
              title: "Questionario sulla salute degli alpinisti",
              text: "",
              plaintext: "<p>Page in Italian</p>"
            },
            {
              id: 81,
              language: {
                id: 1,
                language: "French"
              },
              title: "Enquête générale sur la santé des alpinistes",
              text: "",
              plaintext: "<p>Page in French</p>"
            }
          ],
          children: [],
          deleted: false
        },
        {
          id: 125,
          updated: 1507220001057,
          unactivated: false,
          icon: "fa-pencil",
          position: 3,
          parentId: 126,
          pages_lang: [
            {
              id: 124,
              language: {
                id: 3,
                language: "English"
              },
              title: "Altitude’s acute diseases Survey",
              text: "",
              plaintext: "<p>Page in English</p>"
            },
            {
              id: 125,
              language: {
                id: 1,
                language: "French"
              },
              title: "Sondage sur les maladies aiguës d’altitude",
              text: "",
              plaintext: "<p>Page in French</p>"
            },
            {
              id: 126,
              language: {
                id: 2,
                language: "Italian"
              },
              title: "Questionario sulle malattie acute da altitudine",
              text: "",
              plaintext: "<p>Page in Italian</p>"
            }
          ],
          children: [],
          deleted: false
        },
        {
          id: 127,
          updated: 1507220001068,
          unactivated: false,
          icon: "fa-user",
          position: 4,
          parentId: 126,
          pages_lang: [
            {
              id: 130,
              language: {
                id: 3,
                language: "English"
              },
              title: "Anonymous identifier",
              text: "",
              plaintext: "<p>Page in English</p>"
            },
            {
              id: 132,
              language: {
                id: 2,
                language: "Italian"
              },
              title: "Identificativo anonimo",
              text: "",
              plaintext: "<p>Page in Italian</p>"
            },
            {
              id: 131,
              language: {
                id: 1,
                language: "French"
              },
              title: "Identificateur anonyme",
              text: "",
              plaintext: "<p>Page in French</p>"
            }
          ],
          children: [],
          deleted: false
        }
      ],
      deleted: false
    }
  }
};
