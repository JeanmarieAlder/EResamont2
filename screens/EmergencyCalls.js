import React from "react";
import { Text, View, StyleSheet } from "react-native";
import { globalStyles } from "../styles/global";
import ButtonView from "../components/ButtonView";
import { Linking } from "expo";

export default function EmergencyCalls() {
  let callNumber = number => {
    Linking.openURL("tel:" + number);
  };

  return (
    <View style={localStyles.container}>
      <Text>
        DISCLAIMER: buttons are not working on dev. To test it, it will call
        HES-SO VS at Sierre.
      </Text>
      <Text style={localStyles.headerText}>Europe:</Text>
      <ButtonView
        value={"112"}
        style={localStyles.emergencyButton}
        onPress={() => callNumber("+41586069001")}
      ></ButtonView>
      <View style={globalStyles.topMenuDivider} />
      <Text style={localStyles.headerText}>USA:</Text>
      <ButtonView
        value={"911"}
        style={localStyles.emergencyButton}
        onPress={() => callNumber("+41586069001")}
      ></ButtonView>
      <View style={globalStyles.topMenuDivider} />
      <Text style={localStyles.headerText}>Switzerland:</Text>
      <ButtonView
        value={"144"}
        style={localStyles.emergencyButton}
        onPress={() => callNumber("+41586069001")}
      ></ButtonView>
      <View style={globalStyles.topMenuDivider} />
    </View>
  );
}

const localStyles = StyleSheet.create({
  container: { flex: 1, flexDirection: "column", padding: 15 },
  headerText: {
    fontSize: 30,
    fontWeight: "bold"
  },
  emergencyButton: {
    borderRadius: 10,
    backgroundColor: "red",
    width: "50%",
    height: 40,
    alignItems: "center",
    justifyContent: "center"
  }
});
