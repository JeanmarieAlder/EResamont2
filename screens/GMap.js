import React from "react";
import { View } from "react-native";
import { globalStyles } from "../styles/global";
import { WebView } from "react-native-webview";

export default function GMap({ navigation }) {
  return (
    <View style={globalStyles.mapContainer}>
      <WebView
        source={{
          uri: "https://www.google.com/maps/place/46.539382,%206.69420"
        }}
      ></WebView>
    </View>
  );
}
