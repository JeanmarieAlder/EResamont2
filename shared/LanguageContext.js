import { createContext } from "react";

export const LanguageContext = createContext();
// 1 is French
// 2 is Italian
// 3 is English
// 4 is Deutsch
