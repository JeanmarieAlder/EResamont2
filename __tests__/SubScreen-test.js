import React, { useState } from "react";
import renderer from "react-test-renderer";

import SubScreen from "../screens/SubScreen";
import { checkScreenType } from "../screens/SubScreen";
import { LanguageContext } from "../shared/LanguageContext";
import * as Resources from "../assets/test-pages/Pages";
import storage from "../utils/storage";
import { render, fireEvent } from "react-native-testing-library";

test("SubScreen - Render no child with plaintext", () => {
  const language = 1; //French
  const tree = renderer
    .create(
      <LanguageContext.Provider value={{ language }}>
        <SubScreen navigation={Resources.navigationNoChildPlainText} />
      </LanguageContext.Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test("SubScreen - Render no child with text", () => {
  const language = 1; //French
  const tree = renderer
    .create(
      <LanguageContext.Provider value={{ language }}>
        <SubScreen navigation={Resources.navigationNoChildText} />
      </LanguageContext.Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test("SubScreen - Render with children and text", () => {
  const language = 1; //French
  const tree = renderer
    .create(
      <LanguageContext.Provider value={{ language }}>
        <SubScreen navigation={Resources.navigationWithChildrenAndText} />
      </LanguageContext.Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test("SubScreen - Render with children and no text", () => {
  const language = 1; //French
  const tree = renderer
    .create(
      <LanguageContext.Provider value={{ language }}>
        <SubScreen navigation={Resources.navigationWithChildrenNoText} />
      </LanguageContext.Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test("SubScreen - Render no child in English", () => {
  const language = 3; //English
  const tree = renderer
    .create(
      <LanguageContext.Provider value={{ language }}>
        <SubScreen navigation={Resources.navigationNoChildText} />
      </LanguageContext.Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test("SubScreen - Render no child in Italian", () => {
  const language = 2; //Italian
  const tree = renderer
    .create(
      <LanguageContext.Provider value={{ language }}>
        <SubScreen navigation={Resources.navigationNoChildText} />
      </LanguageContext.Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test("TODO: SubScreen - Render no child in German", () => {
  const language = 1; //TODO: change to 4 (german)
  const tree = renderer
    .create(
      <LanguageContext.Provider value={{ language }}>
        <SubScreen navigation={Resources.navigationNoChildText} />
      </LanguageContext.Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
describe("SubScreen", () => {
  it("generates correct quiz javascript with id 95", () => {
    const language = 2; //Italian
    const tree = renderer
      .create(
        <LanguageContext.Provider value={{ language }}>
          <SubScreen navigation={Resources.navigationId95} />
        </LanguageContext.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("generates correct quiz javascript with id 100", () => {
    const language = 2; //Italian
    const tree = renderer
      .create(
        <LanguageContext.Provider value={{ language }}>
          <SubScreen navigation={Resources.navigationId100} />
        </LanguageContext.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("saves quiz score when message is post from webview", () => {
    const language = 2;

    storage.saveQuizScore = jest.fn();

    let { getByTestId } = render(
      <LanguageContext.Provider value={{ language }}>
        <SubScreen navigation={Resources.navigationId95} />
      </LanguageContext.Provider>
    );
    let webview = getByTestId("sub-webview");
    fireEvent(webview, "onMessage", { nativeEvent: { data: "10" } });
    expect(storage.saveQuizScore).toHaveBeenCalledTimes(1);
  });
  it("navigates to next subscreen when clicking on a children title (with text)", () => {
    const language = 2;
    Resources.navigationWithChildrenAndText.push = jest.fn();
    let { getByText } = render(
      <LanguageContext.Provider value={{ language }}>
        <SubScreen navigation={Resources.navigationWithChildrenAndText} />
      </LanguageContext.Provider>
    );
    let childButton = getByText("Sondaggio di soddisfazione");
    fireEvent(childButton, "onPress");
    expect(Resources.navigationWithChildrenAndText.push).toHaveBeenCalledTimes(
      1
    );
  });
  it("navigates to next subscreen when clicking on a children title (no text)", () => {
    const language = 2;
    Resources.navigationWithChildrenNoText.push = jest.fn();
    let { getByText } = render(
      <LanguageContext.Provider value={{ language }}>
        <SubScreen navigation={Resources.navigationWithChildrenNoText} />
      </LanguageContext.Provider>
    );
    let childButton = getByText("Francia");
    fireEvent(childButton, "onPress");
    expect(Resources.navigationWithChildrenNoText.push).toHaveBeenCalledTimes(
      1
    );
  });
});
