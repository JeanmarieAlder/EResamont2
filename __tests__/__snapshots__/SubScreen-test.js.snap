// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubScreen - Render no child in English 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "html": "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> English page</p>",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "flex": 1,
            "height": 1334,
          },
        ]
      }
      textZoom={270}
      useSharedProcessPool={true}
    />
  </View>
</View>
`;

exports[`SubScreen - Render no child in Italian 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "html": "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> Italian page</p>",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "flex": 1,
            "height": 1334,
          },
        ]
      }
      textZoom={270}
      useSharedProcessPool={true}
    />
  </View>
</View>
`;

exports[`SubScreen - Render no child with plaintext 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 15,
    }
  }
/>
`;

exports[`SubScreen - Render no child with text 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "html": "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> French page</p>",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "flex": 1,
            "height": 1334,
          },
        ]
      }
      textZoom={270}
      useSharedProcessPool={true}
    />
  </View>
</View>
`;

exports[`SubScreen - Render with children and no text 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 20,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "flex-start",
        }
      }
    >
      <View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#232323",
              "borderRadius": 10,
              "height": 40,
              "justifyContent": "center",
              "marginTop": 10,
              "opacity": 1,
              "padding": 10,
              "width": "80%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "sans-serif-medium",
                "textAlign": "center",
                "textAlignVertical": "center",
              }
            }
          >
            France
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#232323",
              "borderRadius": 10,
              "height": 40,
              "justifyContent": "center",
              "marginTop": 10,
              "opacity": 1,
              "padding": 10,
              "width": "80%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "sans-serif-medium",
                "textAlign": "center",
                "textAlignVertical": "center",
              }
            }
          >
            Italie
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#232323",
              "borderRadius": 10,
              "height": 40,
              "justifyContent": "center",
              "marginTop": 10,
              "opacity": 1,
              "padding": 10,
              "width": "80%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "sans-serif-medium",
                "textAlign": "center",
                "textAlignVertical": "center",
              }
            }
          >
            Suisse
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#232323",
              "borderRadius": 10,
              "height": 40,
              "justifyContent": "center",
              "marginTop": 10,
              "opacity": 1,
              "padding": 10,
              "width": "80%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "sans-serif-medium",
                "textAlign": "center",
                "textAlignVertical": "center",
              }
            }
          >
            En Europe
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`SubScreen - Render with children and text 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <RNCWebView
        cacheEnabled={true}
        javaScriptEnabled={true}
        messagingEnabled={false}
        onContentProcessDidTerminate={[Function]}
        onHttpError={[Function]}
        onLoadingError={[Function]}
        onLoadingFinish={[Function]}
        onLoadingProgress={[Function]}
        onLoadingStart={[Function]}
        onMessage={[Function]}
        onShouldStartLoadWithRequest={[Function]}
        source={
          Object {
            "html": "<p>Text in French.  Also it's located in plaintext field.</p>",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
            Object {
              "flex": 1,
              "height": 333.5,
            },
          ]
        }
        textZoom={270}
        useSharedProcessPool={true}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "flex-start",
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#232323",
            "borderRadius": 10,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "opacity": 1,
            "padding": 10,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "sans-serif-medium",
              "textAlign": "center",
              "textAlignVertical": "center",
            }
          }
        >
          Questionnaire de satisfaction
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#232323",
            "borderRadius": 10,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "opacity": 1,
            "padding": 10,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "sans-serif-medium",
              "textAlign": "center",
              "textAlignVertical": "center",
            }
          }
        >
          Enquête générale sur la santé des alpinistes
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#232323",
            "borderRadius": 10,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "opacity": 1,
            "padding": 10,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "sans-serif-medium",
              "textAlign": "center",
              "textAlignVertical": "center",
            }
          }
        >
          Sondage sur les maladies aiguës d’altitude
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#232323",
            "borderRadius": 10,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "opacity": 1,
            "padding": 10,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "sans-serif-medium",
              "textAlign": "center",
              "textAlignVertical": "center",
            }
          }
        >
          Identificateur anonyme
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TODO: SubScreen - Render no child in German 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "html": "<p><b>ALTIDOC - Téléconsultation et médecine de montagne</b> French page</p>",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "flex": 1,
            "height": 1334,
          },
        ]
      }
      textZoom={270}
      useSharedProcessPool={true}
    />
  </View>
</View>
`;
